// @adins/uc-subsection/bundles/uc-subsection.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/platform-browser'), require('@angular/core'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('uc-subsection', ['exports', '@angular/platform-browser', '@angular/core', '@angular/common'], factory) :
    (factory((global['uc-subsection'] = {}),global.ng.platformBrowser,global.ng.core,global.ng.common));
}(this, (function (exports,platformBrowser,i0,common) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var UcSubsectionService = /** @class */ (function () {
        function UcSubsectionService() {
        }
        UcSubsectionService.decorators = [
            { type: i0.Injectable, args: [{
                        providedIn: 'root'
                    },] }
        ];
        /** @nocollapse */
        UcSubsectionService.ctorParameters = function () { return []; };
        /** @nocollapse */ UcSubsectionService.ngInjectableDef = i0.defineInjectable({ factory: function UcSubsectionService_Factory() { return new UcSubsectionService(); }, token: UcSubsectionService, providedIn: "root" });
        return UcSubsectionService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var UcSubsectionComponent = /** @class */ (function () {
        function UcSubsectionComponent(_renderer2, _document) {
            this._renderer2 = _renderer2;
            this._document = _document;
            this.isHidden = false;
        }
        /**
         * @return {?}
         */
        UcSubsectionComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                console.log(this.title);
                console.log(this.panel);
                /** @type {?} */
                var js = this._renderer2.createElement('script');
                js.text = "\n          $(document).ready(function(){\n            $(\"#" + this.id + "\").click(function(){\n              $(\"#" + this.panel + "\").slideToggle(\"slow\");\n            });\n          });\n        ";
                this._renderer2.appendChild(this._document.body, js);
            };
        UcSubsectionComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'lib-ucSubsection',
                        template: "<div class=\"card-body\">\r\n  <form class=\"form form-horizontal\">\r\n    <div class=\"form-body\">\r\n      <h4 class=\"form-section font-weight-bold\">\r\n        <div [id]=\"id\" class=\"btn no-padding cursor-pointer\" (click)=\"isHidden = !isHidden\">\r\n          <i class=\"fa\" style=\"font-size: 15px; margin: 0px 0px 5px -15px;\"\r\n            [ngClass]=\"isHidden ? 'fa-chevron-right' : 'fa-chevron-down'\"></i>\r\n        </div> {{title}}\r\n      </h4>\r\n    </div>\r\n  </form>\r\n</div>"
                    }] }
        ];
        /** @nocollapse */
        UcSubsectionComponent.ctorParameters = function () {
            return [
                { type: i0.Renderer2 },
                { type: undefined, decorators: [{ type: i0.Inject, args: [platformBrowser.DOCUMENT,] }] }
            ];
        };
        UcSubsectionComponent.propDecorators = {
            title: [{ type: i0.Input }],
            panel: [{ type: i0.Input }],
            id: [{ type: i0.Input }]
        };
        return UcSubsectionComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var UcSubsectionModule = /** @class */ (function () {
        function UcSubsectionModule() {
        }
        UcSubsectionModule.decorators = [
            { type: i0.NgModule, args: [{
                        declarations: [UcSubsectionComponent],
                        imports: [
                            common.CommonModule
                        ],
                        exports: [UcSubsectionComponent]
                    },] }
        ];
        return UcSubsectionModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    exports.UcSubsectionService = UcSubsectionService;
    exports.UcSubsectionComponent = UcSubsectionComponent;
    exports.UcSubsectionModule = UcSubsectionModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
