// @adins/ucgridfooter/bundles/ucgridfooter.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@ng-bootstrap/ng-bootstrap'), require('@angular/forms'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('ucgridfooter', ['exports', '@angular/core', '@ng-bootstrap/ng-bootstrap', '@angular/forms', '@angular/common'], factory) :
    (factory((global.ucgridfooter = {}),global.ng.core,global.ngBootstrap,global.ng.forms,global.ng.common));
}(this, (function (exports,i0,ngBootstrap,forms,common) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var UcgridfooterService = /** @class */ (function () {
        function UcgridfooterService() {
        }
        UcgridfooterService.decorators = [
            { type: i0.Injectable, args: [{
                        providedIn: 'root'
                    },] }
        ];
        /** @nocollapse */
        UcgridfooterService.ctorParameters = function () { return []; };
        /** @nocollapse */ UcgridfooterService.ngInjectableDef = i0.defineInjectable({ factory: function UcgridfooterService_Factory() { return new UcgridfooterService(); }, token: UcgridfooterService, providedIn: "root" });
        return UcgridfooterService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var UcgridfooterComponent = /** @class */ (function () {
        function UcgridfooterComponent() {
            this.select = new i0.EventEmitter();
        }
        /**
         * @return {?}
         */
        UcgridfooterComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () {
                console.log("Grid Footer Init");
                this.pageNow = 1;
                this.pageSize = 10;
            };
        /**
         * @param {?} event
         * @return {?}
         */
        UcgridfooterComponent.prototype.searchPagination = /**
         * @param {?} event
         * @return {?}
         */
            function (event) {
                console.log("Grid Footer Page Change");
                this.pageNow = event;
                /** @type {?} */
                var paging = { pageNow: event, pageSize: this.pageSize };
                this.select.emit(paging);
            };
        /**
         * @return {?}
         */
        UcgridfooterComponent.prototype.onChange = /**
         * @return {?}
         */
            function () {
                console.log("Grid Footer Page Change");
                /** @type {?} */
                var paging = { pageNow: this.pageNow, pageSize: this.pageSize };
                this.select.emit(paging);
            };
        UcgridfooterComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'lib-ucgridfooter',
                        template: "<div class=\"gridFooter row col-md-12\">\r\n    <div class=\"col-md-4 col-xs-12\">\r\n        <select class=\"custom-select\" style=\"width: auto;\" name=\"pageSize\" [(ngModel)]=\"pageSize\"\r\n            (change)=\"onChange()\">\r\n            <option [ngValue]=\"10\">10</option>\r\n            <option [ngValue]=\"20\">20</option>\r\n            <option [ngValue]=\"50\">50</option>\r\n        </select>\r\n    </div>\r\n    <div class=\"justify-content-center col-md-5 col-xs-12\">\r\n        <ngb-pagination (pageChange)=\"searchPagination($event)\" [pageSize]=\"pageSize\" [collectionSize]=\"totalData\"\r\n            [(page)]=\"pageNow\" [maxSize]=\"5\" [rotate]=\"true\" [ellipses]=\"true\" [boundaryLinks]=\"false\"></ngb-pagination>\r\n\r\n    </div>\r\n    <div class=\"col-md-3 col-xs-12\" style=\"font-weight: bold; text-align: right\" translate>Total Data : {{totalData}}</div>\r\n</div>"
                    }] }
        ];
        /** @nocollapse */
        UcgridfooterComponent.ctorParameters = function () { return []; };
        UcgridfooterComponent.propDecorators = {
            select: [{ type: i0.Output }]
        };
        return UcgridfooterComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var UcgridfooterModule = /** @class */ (function () {
        function UcgridfooterModule() {
        }
        UcgridfooterModule.decorators = [
            { type: i0.NgModule, args: [{
                        declarations: [UcgridfooterComponent],
                        imports: [
                            common.CommonModule,
                            ngBootstrap.NgbModule,
                            forms.FormsModule
                        ],
                        exports: [UcgridfooterComponent, ngBootstrap.NgbModule]
                    },] }
        ];
        return UcgridfooterModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    exports.UcgridfooterService = UcgridfooterService;
    exports.UcgridfooterComponent = UcgridfooterComponent;
    exports.UcgridfooterModule = UcgridfooterModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
