// @adins/uc-subsection/esm2015/lib/uc-subsection.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, Renderer2, Inject } from '@angular/core';
import { DOCUMENT } from '@angular/platform-browser';
export class UcSubsectionComponent {
    /**
     * @param {?} _renderer2
     * @param {?} _document
     */
    constructor(_renderer2, _document) {
        this._renderer2 = _renderer2;
        this._document = _document;
        this.isHidden = false;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        console.log(this.title);
        console.log(this.panel);
        /** @type {?} */
        let js = this._renderer2.createElement('script');
        js.text = `
          $(document).ready(function(){
            $("#` + this.id + `").click(function(){
              $("#` + this.panel + `").slideToggle("slow");
            });
          });
        `;
        this._renderer2.appendChild(this._document.body, js);
    }
}
UcSubsectionComponent.decorators = [
    { type: Component, args: [{
                selector: 'lib-ucSubsection',
                template: "<div class=\"card-body\">\r\n  <form class=\"form form-horizontal\">\r\n    <div class=\"form-body\">\r\n      <h4 class=\"form-section font-weight-bold\">\r\n        <div [id]=\"id\" class=\"btn no-padding cursor-pointer\" (click)=\"isHidden = !isHidden\">\r\n          <i class=\"fa\" style=\"font-size: 15px; margin: 0px 0px 5px -15px;\"\r\n            [ngClass]=\"isHidden ? 'fa-chevron-right' : 'fa-chevron-down'\"></i>\r\n        </div> {{title}}\r\n      </h4>\r\n    </div>\r\n  </form>\r\n</div>"
            }] }
];
/** @nocollapse */
UcSubsectionComponent.ctorParameters = () => [
    { type: Renderer2 },
    { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] }] }
];
UcSubsectionComponent.propDecorators = {
    title: [{ type: Input }],
    panel: [{ type: Input }],
    id: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    UcSubsectionComponent.prototype.title;
    /** @type {?} */
    UcSubsectionComponent.prototype.panel;
    /** @type {?} */
    UcSubsectionComponent.prototype.id;
    /** @type {?} */
    UcSubsectionComponent.prototype.isHidden;
    /**
     * @type {?}
     * @private
     */
    UcSubsectionComponent.prototype._renderer2;
    /**
     * @type {?}
     * @private
     */
    UcSubsectionComponent.prototype._document;
}
