// @adins/uc-subsection/fesm2015/uc-subsection.js
import { DOCUMENT } from '@angular/platform-browser';
import { Injectable, Component, Input, Renderer2, Inject, NgModule, defineInjectable } from '@angular/core';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class UcSubsectionService {
    constructor() { }
}
UcSubsectionService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
UcSubsectionService.ctorParameters = () => [];
/** @nocollapse */ UcSubsectionService.ngInjectableDef = defineInjectable({ factory: function UcSubsectionService_Factory() { return new UcSubsectionService(); }, token: UcSubsectionService, providedIn: "root" });

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class UcSubsectionComponent {
    /**
     * @param {?} _renderer2
     * @param {?} _document
     */
    constructor(_renderer2, _document) {
        this._renderer2 = _renderer2;
        this._document = _document;
        this.isHidden = false;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        console.log(this.title);
        console.log(this.panel);
        /** @type {?} */
        let js = this._renderer2.createElement('script');
        js.text = `
          $(document).ready(function(){
            $("#` + this.id + `").click(function(){
              $("#` + this.panel + `").slideToggle("slow");
            });
          });
        `;
        this._renderer2.appendChild(this._document.body, js);
    }
}
UcSubsectionComponent.decorators = [
    { type: Component, args: [{
                selector: 'lib-ucSubsection',
                template: "<div class=\"card-body\">\r\n  <form class=\"form form-horizontal\">\r\n    <div class=\"form-body\">\r\n      <h4 class=\"form-section font-weight-bold\">\r\n        <div [id]=\"id\" class=\"btn no-padding cursor-pointer\" (click)=\"isHidden = !isHidden\">\r\n          <i class=\"fa\" style=\"font-size: 15px; margin: 0px 0px 5px -15px;\"\r\n            [ngClass]=\"isHidden ? 'fa-chevron-right' : 'fa-chevron-down'\"></i>\r\n        </div> {{title}}\r\n      </h4>\r\n    </div>\r\n  </form>\r\n</div>"
            }] }
];
/** @nocollapse */
UcSubsectionComponent.ctorParameters = () => [
    { type: Renderer2 },
    { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] }] }
];
UcSubsectionComponent.propDecorators = {
    title: [{ type: Input }],
    panel: [{ type: Input }],
    id: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class UcSubsectionModule {
}
UcSubsectionModule.decorators = [
    { type: NgModule, args: [{
                declarations: [UcSubsectionComponent],
                imports: [
                    CommonModule
                ],
                exports: [UcSubsectionComponent]
            },] }
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { UcSubsectionService, UcSubsectionComponent, UcSubsectionModule };
