// @adins/ucsearch/esm2015/lib/model/RequestCriteriaObj.model.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
export class RequestCriteriaObj {
    constructor() {
        this.includeCount = true;
        this.includeData = true;
        this.isLoading = true;
        this.queryString = '';
        this.rowVersion = '';
    }
}
if (false) {
    /** @type {?} */
    RequestCriteriaObj.prototype.includeCount;
    /** @type {?} */
    RequestCriteriaObj.prototype.includeData;
    /** @type {?} */
    RequestCriteriaObj.prototype.pageNo;
    /** @type {?} */
    RequestCriteriaObj.prototype.rowPerPage;
    /** @type {?} */
    RequestCriteriaObj.prototype.orderBy;
    /** @type {?} */
    RequestCriteriaObj.prototype.criteria;
    /** @type {?} */
    RequestCriteriaObj.prototype.isLoading;
    /** @type {?} */
    RequestCriteriaObj.prototype.queryString;
    /** @type {?} */
    RequestCriteriaObj.prototype.rowVersion;
}
